File: app/backend/java/smartride-ai-service/src/main/java/com/smartride/smartride_ai_service/dto/ChatResponse.java
package com.smartride.smartride_ai_service.dto;

/**
 * Response DTO for the AI chat endpoint.
 * Returns the AI-generated response text.
 */
public class ChatResponse {

    private String response;

    public ChatResponse() {}

    public ChatResponse(String response) {
        this.response = response;
    }

    // --- Getter & Setter ---

    public String getResponse() {
        return response;
    }

    public void setResponse(String response) {
        this.response = response;
    }
}

File: app/backend/.net/SmartRide.AiService/Models/ChatResponse.cs
using System.Text.Json.Serialization;

namespace SmartRide.AiService.Models;

/// <summary>
/// Response DTO for the SmartRide AI Chat endpoint.
/// </summary>
public class ChatResponse
{
    [JsonPropertyName("response")]
    public string Response { get; set; } = string.Empty;

    [JsonPropertyName("sessionId")]
    public string? SessionId { get; set; }

    [JsonPropertyName("timestamp")]
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;

    public ChatResponse() { }

    public ChatResponse(string response, string? sessionId = null)
    {
        Response = response;
        SessionId = sessionId;
    }
}
